### PR TITLE
Simulated Analytics Players

### PR DESCRIPTION
**Pull request by krist00fer**
### Issue: #486 #528
 - [x] Tick here to confirm that you have run RunCodeFormatter.ps1

### Description:
AnalyticsTestClient has been cleaned up and functionality added. You can now:

* Generate random geo-routes throughout the world that can then be followed by virtual walking players, GeoBots.
* Simulate virtual users over a long period of time and send tens of thousands of messages to EventHub to create accurate test data.

### Test:

1. To generate routes, you first need to download the Open Street Map data file for the region where you want to generate routes for. You can download the correct file for your area [here] (http://planet.openstreetmap.org/). Save the .osm.pbf file somewhere locally on your computer.

2. Use a tool like [geojson.io](http://geojson.io) to figure out the corner coordinates for the area where you want to generate routes for.

3. Start the AnalyticsTestClient and select option 4 and then 1 to generate routes.

4. Answer the questions according to your setup and will

5. Watch how your harddrive is filled with generated routes. Open a random file and copy the content to clipboard and then paste it onto [geojson.io](http://geojson.io) to see the route on top of a map.

Keep this files on your harddrive for later use.

6. Now it's time to generate data with the routes you have. In AnalyticsTestClient, select option 4 and 2.

7. Answer the questions according to your will. Notice that quite a lot of data will be generated in a very short time, so keep the timespan short if you don't want to generate a lot of data.

8. Make sure you also start the Nether.Analytics.Host project (even though you don't need to have them running at the same time).

9. Look at the folderstructure that should be created on your harddrive as well in ADFS in the cloud.

10. Pick a random file and make sure it looks according to what you expected it to look like.

(Make sure that all above information is provided before submitting your pull request)

### CODE REVIEW

**Comment by krist00fer** [comment]
A tip is to create an EventHub with many partitions since the reading from it goes substantially faster then.

File: src/TestClients/AnalyticsTestClient/Bots/NetherBot.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        {
+            return now - StartTime > _options.SessionLenght;
+        }
+    }
+
+    public class NetherBotOptions
+    {
+        public bool SessionStartMsg { get; set; } = true;
+        public TimeSpan HeartBeatMsgInterval { get; set; } = TimeSpan.FromMinutes(1);
+        public TimeSpan SessionLenght { get; set; } = RandomEx.TimeSpan(TimeSpan.FromSeconds(5), TimeSpan.FromMinutes(20));
```
Comment (stuartleeks) [inline src/TestClients/AnalyticsTestClient/Bots/NetherBot.cs]: typo ;-)

File: src/TestClients/AnalyticsTestClient/Clients/INetherAnalyticsClient.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+using Nether.Analytics.MessageFormats;
+using System;
+using System.Threading.Tasks;
+
+namespace AnalyticsTestClient
+{
+    public interface IAnalyticsClient
+    {
+        Task SendMsgAsync(INetherMessage msg, DateTime? dbgEnqueuedTimeUtc = null);
```
Comment (stuartleeks) [inline src/TestClients/AnalyticsTestClient/Clients/INetherAnalyticsClient.cs]: Would prefer `SendMessageAsync`

File: src/TestClients/AnalyticsTestClient/Menus/SimulateMovementMenu.cs [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+                if (Console.KeyAvailable)
+                {
+                    if (Console.ReadKey().Key == ConsoleKey.Escape)
+                    {
+                        break;
+                    }
+                }
+            }
+
+            //Parallel.For(0, noRoutes, (i, s) =>
```
Comment (stuartleeks) [inline src/TestClients/AnalyticsTestClient/Menus/SimulateMovementMenu.cs]: Can this code be removed?

File: src/TestClients/AnalyticsTestClient/Menus/SimulateMovementMenu.cs [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+                    Console.WriteLine("Loading Open Street Map Database. This can take a while!");
+                    var loadOsmDataTask = Task.Run(() =>
+                    {
+                        s_routerDb.LoadOsmData(stream, Vehicle.Pedestrian);
+                    });
+
+                    while (!loadOsmDataTask.IsCompleted)
+                    {
+                        Console.Write(".");
+                        Task.Delay(1000).Wait();
```
Comment (stuartleeks) [inline src/TestClients/AnalyticsTestClient/Menus/SimulateMovementMenu.cs]: Could just use Thread.Sleep here ;-)

Comment (krist00fer) [inline src/TestClients/AnalyticsTestClient/Menus/SimulateMovementMenu.cs]: Ok, will do

File: src/TestClients/AnalyticsTestClient/Menus/SimulateMovementMenu.cs (hunk 3 above)
Comment (krist00fer) [inline src/TestClients/AnalyticsTestClient/Menus/SimulateMovementMenu.cs]: This commented out section actually works and runs faster but gives you somewhat more problem with debugging. That's why I left it there, but did comment it out.

File: src/TestClients/AnalyticsTestClient/Bots/NetherBot.cs (hunk 1 above)
Comment (krist00fer) [inline src/TestClients/AnalyticsTestClient/Bots/NetherBot.cs]: Fixed typo Lenght to Length

File: src/TestClients/AnalyticsTestClient/Clients/INetherAnalyticsClient.cs (hunk 2 above)
Comment (krist00fer) [inline src/TestClients/AnalyticsTestClient/Clients/INetherAnalyticsClient.cs]: I changed all use of Msg to Message

**Pull request merged**